compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -585,6 +587,12 @@ public void visit(BLangIndexBasedAccess indexAccessExpr) {
     @Override
     public void visit(BLangInvocation invocationExpr) {
         analyzeNode(invocationExpr.expr, env);
+        if (!isFieldsInitializedForSelfKeyWordExpression(invocationExpr)) {
```

Comment:
The method names are a bit misleading IMO :) Shall we say

`isFieldsInitializedForSelfArgument`

and

`isFieldsInitializedForSelfInvocation`

Reply:
Changed